src/main/java/de/bonndan/nivio/input/ItemDescriptionValues.java

```diff
@@ -13,32 +16,46 @@
      */
     public static void assignNotNull(ItemDescription existing, ItemDescription increment) {
 
-        if (increment.getName() != null)
-            existing.setName(increment.getName());
+        ComponentDescriptionValues.assignNotNull(existing, increment);
+
         if (increment.getType() != null)
             existing.setType(increment.getType());
-        if (increment.getDescription() != null)
-            existing.setDescription(increment.getDescription());
         if (increment.getContact() != null)
             existing.setContact(increment.getContact());
 
-        if (increment.getOwner() != null)
-            existing.setOwner(increment.getOwner());
         if (increment.getGroup() != null)
             existing.setGroup(increment.getGroup());
 
-        if (increment.getLabels() != null) {
-            increment.getLabels().forEach((s, s2) -> {
-                if (increment.getLabel(s) != null) {
-                    existing.setLabel(s, s2);
-                }
-            });
-        }
-
-        existing.getLinks().putAll(increment.getLinks());
-
         assignSafe(increment.getRelations(), (rel) -> rel.forEach(existing::addRelation));
 
         assignSafe(increment.getInterfaces(), (set) -> set.forEach(intf -> existing.getInterfaces().add(intf)));
     }
+
+    /**
+     * Writes the values of the template (second object) to the first where first is null.
+     *
+     * @param item     target
+     * @param template source
+     */
+    public static void assignSafeNotNull(ItemDescription item, ItemDescription template) {
+
+        ComponentDescriptionValues.assignSafeNotNull(item, template);
+
+        assignSafeIfAbsent(template.getType(), item.getType(), item::setType);
+        assignSafeIfAbsent(template.getContact(), item.getContact(), item::setContact);
+        assignSafeIfAbsent(template.getGroup(), item.getGroup(), item::setGroup);
+
+        if (template.getProvidedBy() != null) {
+            template.getProvidedBy().stream()
+                    .filter(s -> !StringUtils.isEmpty(s) && !item.getProvidedBy().contains(s))
+                    .forEach(s -> item.getProvidedBy().add(s));
+        }
+
+        template.getRelations().forEach(item::addRelation);
+
+        template.getInterfaces().forEach(interfaceItem -> {
+            if (!item.getInterfaces().contains(interfaceItem))
+                item.getInterfaces().add(interfaceItem);
+        });
```

Comment:
The contains check is not necessary. Can't this also be simpler like this: `item.getInterfaces().addAll(template.getInterfaces());`?

Reply:
done